src/node_dotenv.cc

```diff
@@ -0,0 +1,121 @@
+#include "env-inl.h"
+#include "node_dotenv.h"
+#include "node_file.h"
+#include "uv.h"
+
+#ifdef __POSIX__
+constexpr std::string_view kNewlineCharacter = "\n";
+#else
+constexpr std::string_view kNewlineCharacter = "\r\n";
```

Comment:
I think it is always `\r\n` in Windows. @nodejs/platform-windows can you confirm?

Reply:
I think you’re already excluding leading/trailing whitespace for each line, outside of quotes? If `\r` is considered a whitespace character then you can break on `\n` and be fine. We should probably have a test for this.

Reply:
> I think it is always `\r\n` in Windows. @nodejs/platform-windows can you confirm?

Not at all. While it's true that some programs will default to CRLF on Windows with new text files, most software support LF (even notepad.exe), and a lot can be configured to use it by default. For example, if I create a new text file on Windows with VS Code, it will have LF.

Reply:
Thanks. I've updated my implementation.